### PR TITLE
Deprecate `config` in `create_cnn` and instead pass kwargs directly

### PR DESCRIPTION
**Pull request by borisdayma**
`cnn_learner` and `unet_learner` have now direct access to all config parameters and don't require a separate `config` object anymore.

### CODE REVIEW

**Comment by borisdayma** [comment]
Now I'm getting more params logged than before!

The next step will be to figure out how to log params from `learn.fit`, `learn.fit_one_cycle`, etc… (included method name).

**Comment by borisdayma** [comment]
It should now be backwards compatible

File: fastai/vision/learner.py [hunk 1]
```diff
@@ -161,42 +156,76 @@ def _add_norm(dls, meta, pretrained):
...
+                # learner args
+                loss_func=None, opt_func=Adam, lr=defaults.lr, splitter=None, cbs=None, metrics=None, path=None,
+                model_dir='models', wd=None, wd_bn_bias=False, train_bn=True, moms=(0.95,0.85,0.95),
+                # other model args
+                **kwargs):
     "Build a convnet style learner from `dls` and `arch`"
-    if config is None: config = {}
+
+    if config:
+        print('Warning: config is deprecated and will be removed in the future. Pass your args directly to cnn_learner.')
```
Comment (jph00) [inline fastai/vision/learner.py:169]:
```suggestion
        warnings.warn('config param is deprecated. Pass your args directly to cnn_learner.')
```

**Comment by borisdayma** [comment]
The CI failure seems unrelated

**Comment by jph00** [comment]
Thank you!

**Pull request merged**